source/server/active_tcp_socket.cc

```diff
@@ -16,7 +16,11 @@ ActiveTcpSocket::ActiveTcpSocket(ActiveStreamListenerBase& listener,
       iter_(accept_filters_.end()),
       stream_info_(std::make_unique<StreamInfo::StreamInfoImpl>(
           listener_.dispatcher().timeSource(), socket_->addressProviderSharedPtr(),
-          StreamInfo::FilterState::LifeSpan::Connection)) {
+          StreamInfo::FilterState::LifeSpan::Connection)),
+      inspect_buffer_(listener_.dispatcher().getWatermarkFactory().createBuffer(
+          []() -> void { /* TODO(soulxu) handle watermark*/ },
```

Comment:
Because this buffer is used in a place where we can't be streaming data, and there's a limit on how much can be read (provided by listener filters), I think a non-watermark buffer may be better here.

Reply:
got it